sled-agent/src/services.rs

```diff
@@ -709,6 +828,71 @@ impl ServiceManager {
 
         Ok(())
     }
+
+    pub async fn ensure_switch(&self, request: Option<ServiceZoneRequest>) {
```

Comment:
This is worth taking a look at - I wanted the "hardware monitoring" portion of the sled agent to be able to tell the "service managing" portion of sled agent that "if you can, you should keep trying to launch the switch zone until I tell you otherwise".

As a result, I've structured this as a bit of a state machine. The `ServiceManager` spawns a background task that keeps trying to bring up the switch zone until it has started successfully. At any point, the hardware monitor may see that the tofino driver is unloaded, and call `ensure_switch(None)`, which basically means "stop what you're doing, ensure the switch zone does not exist".